src/99-appendices/04-entity-table.md

```diff
@@ -10,23 +10,23 @@ file type are denoted. Entity formats indicate whether the value is alphanumeric
-| headshape<br>(meg)                                                                             | REQUIRED      | OPTIONAL      |                | OPTIONAL      |                          |                |                          |               |                        |                |                     |                       | OPTIONAL        |
-| markers<br>(meg)                                                                               | REQUIRED      | OPTIONAL      | OPTIONAL       | OPTIONAL      |                          |                |                          |               |                        |                |                     |                       | OPTIONAL        |
-| photo<br>(meg/eeg/ieeg)                                                                        | REQUIRED      | OPTIONAL      |                | OPTIONAL      |                          |                |                          |               |                        |                |                     |                       |                 |        
-| electrodes<br>(eeg/ieeg)                                                                       | REQUIRED      | OPTIONAL      |                | OPTIONAL      |                          |                |                          |               |                        |                |                     |                       | OPTIONAL        |
-| events<br>(meg/eeg/ieeg)                                                                       | REQUIRED      | OPTIONAL      | REQUIRED       |               |                          |                |                          | OPTIONAL      |                        |                |                     |                       |                 |        
+| Entity                                                                                         | Subject       | Session       | Task           | Acquisition   | Contrast Enhancing Agent | Reconstruction | Phase-Encoding Direction | Run           | Corresponding modality | Echo           | Recording           | Processed (on device) | Space           | Part           |
+| :--------------------------------------------------------------------------------------------- | :------------ | :------------ | :------------- | :------------ | :----------------------- | :------------- | :----------------------- | :------------ | :--------------------- | :------------- | :------------------ | :-------------------- | :---------------| :------------- |
+| Format                                                                                         | `sub-<label>` | `ses-<label>` | `task-<label>` | `acq-<label>` | `ce-<label>`             | `rec-<label>`  | `dir-<label>`            | `run-<index>` | `mod-<label>`          | `echo-<index>` | `recording-<label>` | `proc-<label>`        | `space-<label>` | `part-<index>` |
```

Comment:
I had not interpreted `<index>` as index into something explicitly (as in expressing a link), rather as how you would use it in a for-loop, which means a sequence of integers that increment by one. The starting value is not defined AFAIK, and I would expect it either to be 0 or 1.

The [run entity](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#the-run-entity) is the earliest use of `<index>`, but it is also used (in the same part of the spec) for the echo entity (although not explicitly defined there, it is in examples and in the entity appendix table).

The use of `part-` in MEG requires it to be `<index>` and that is also how it is specified now, so I think the fix in this PR is correct (as long as `<id>` is changed to `<index>`). Extending the value to `<label>` would be another discussion. Right now I don't think we have consensus yet on re-using part (with a different meaning of the entity) for non-MEG data, so I think it would be too early to decide for loosening it up from index to label.

Reply:
Re `<index>` as an explicit index, it might be coming as a part of BEP001 (ideally should be a separate BEP or a PR first): https://github.com/bids-standard/bep001/blob/indexable_metadata/src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md#the-indexable_metadata-entity . ATM we use it only for run and echo.  Indeed the value of `<index>` is not yet clearly defined/controlled in BIDS otherwise.

Reply:
> The use of part- in MEG requires it to be <index> ...

That was my question really in [above comment](https://github.com/bids-standard/bids-specification/pull/460#discussion_r420218216) -- is it required?  For clarity: is the only tool in question (mne-python) relies on that index, or users would just point to the first file in the sequence and then internal pointer to the next file is used?  I saw use of `_construct_bids_filename` (the only place I found `_part-` mentioned) only within `write_data` but have not spotted it elsewhere (somewhere in reading).

Reply:
Internal pointing can IMHO not be relied on since for most users it will not be possible (i.e. they do not have the tools) to update these internal pointers when they rename the file. AFAIK the only tool that can be used for updating these internal pointers is MNE-Python. I do not recall ever seeing such a tool from the vendor itself. 
Putting the responsibility with the user to provide the files in the right sequence (as we e.g. do in FIeldTrip) would always be possible.